model/template.go

```diff
@@ -110,9 +249,23 @@ AND catalog_template.base = ?`
 
 	db.Raw(query, params...).Find(&templateModels)
 
+	var templateIDList []int
+	for _, template := range templateModels {
+		templateIDList = append(templateIDList, int(template.ID))
+	}
+
+	templateCategoryMap := templateCategoryMap(db, templateIDList)
```

Comment:
Nitpick: You don't really need to change this, but it is confusing to have method and variables named the same thing like this. Generally, my preference is for functions like this to start with a verb like fetchTemplateCategories or generateTemplateCategories